test/test-mongocrypt-traverse-util.c

```diff
@@ -116,28 +116,33 @@ static void _assert_correct_fields(bson_t *bson,
 
     if (tester->parent != NEST_IN_NONE) {
         for (i = 0; i < num_other; i++) {
-            snprintf(name, 13, "other.field%d", i);
+            /* use mod 10, since guaranteed < 10 and to silence -Wformat-truncation */
+            snprintf(name, 13, "other.field%d", i % 10);
             bson_iter_init(&iter, bson);
             BSON_ASSERT(bson_iter_find_descendant(&iter, name, &child));
```

Comment:
I expect there is no need to reuse `name` since this is test code.

Consider using `bson_strdup_printf` to allocate the keys in these tests to avoid the `mod 10` workaround:

```c
        for (i = 0; i < num_other; i++) {
            char *key = bson_strdup_printf("other.field%d", i);
            bson_iter_init(&iter, bson);
            BSON_ASSERT(bson_iter_find_descendant(&iter, key, &child));
            bson_free(key);
        }
```

Reply:
Oh, that's a good idea. It definitely did not occur to me. Change incoming.